### DISCUSSION TITLE
Use player name instead of uuid

### DISCUSSION THREAD

**Discussion by akex06** [GriefPrevention/GriefPrevention#1661] [General]
Is there a way of using player name instead of uuid in the saving of the lands?

**Comment by bigpresh** [comment]
As in writing the player name to the claim data files?  No - and it'd generally be unwise, since if the player renames their account, they'd have no claims when they next log in (and if someone else then renamed their account to that player's old name, after whatever the timeout is, they'd then have access to all their claims).

**Reply by akex06** [reply]
I would need to do it this way because I want to make my server Java instead of pocketmine, the problem is that the old plugin that I used in pocketmine used names (which ones most of them aren't paid accounts and also are bedrock accounts), so for moving from pocketmine to bedrock I would need to do it that way

**Reply by bigpresh** [reply]
Well, GP will use UUIDs, but you may be able to come up with some way to fake it - after all, it works for offline servers where there's no real UUID for each name, so *something* gets generated by the server and put in `usercache.json`.

I'm not entirely clear if you're trying to migrate some existing claims from your "pocketmine" server to a new Java server.  You also say you want to "make my server Java", but then "moving from pocketmine to bedrock" - presumably the latter is just a mistake?

If it's just a case of "I want it to support offline mode i.e. non-paid accounts", it should work - it does for me, admittedly on an internal server with only 4 players, one of whom is a paid account and the other 3 are not.
